src/core/id-headers.js

```diff
@@ -13,9 +13,13 @@ export function run(conf) {
     `section:not(.head):not(.introductory) h2, h3, h4, h5, h6`
   );
   for (const h of headings) {
-    addId(h);
+    // prefer for ID: heading.id > parentElement.id > newly generated heading.id
+    let id = h.id;
```

Comment:
I think we still want to id every heading regardless (I think it might be a W3C PubRules requirement), even if we prefer the other.

Reply:
We are adding an id if it doesn't exist (next line)

Reply:
resolving based on discussions in slack

Reply:
<details>
<summary>Conversation log (in case we forget what we discussed)</summary>
<pre>
Marcos:  do we need `let id = h.id`;
sidvishnoi: we don't have a way to know if it's already existing id or newly added one (so we test it first) ... so "preference"</pre>
</details>